autoload/vader.vim

```diff
@@ -38,10 +38,21 @@ function! vader#run(bang, ...) range
     let [line1, line2] = [1, 0]
   endif
 
-  if a:0 == 0
+  let args = copy(a:000)
+  let patterns = []
+  let options = {
+        \ 'exitfirst': 0,
+        \ }
+  while len(args)
+    let arg = remove(args, 0)
+    if arg ==# '-x'
+      let options.exitfirst = 1
+    else
+      let patterns += [arg]
+    endif
+  endwhile
```

Comment:
Since we only have one option, the following should suffice:

```vim
let options = { 'exitfirst': index(a:000, '-x') >= 0 }
let patterns = filter(copy(a:000), 'v:val != "-x"')
```